--- pic30-binutils/c30_resource/src/c30/c30_flag_definitions.h.orig	2011-11-28 12:09:19.000000000 +0000
+++ pic30-binutils/c30_resource/src/c30/c30_flag_definitions.h	2011-11-28 19:08:27.000000000 +0000
@@ -81,34 +81,38 @@
  *   '/' is converted to the correct directory separator for the OS
  */
 #define MPLABC30_PIC30F_INC_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "dsPIC30F" DIR_SEPARATOR_STR \
                                  "inc"
-#define MPLABC30_PIC33E_INC_PATH ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "dsPIC33E" DIR_SEPARATOR_STR \
-                                 "inc"
 #define MPLABC30_PIC33F_INC_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "dsPIC33F" DIR_SEPARATOR_STR \
                                  "inc"
 #define MPLABC30_PIC33E_INC_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "dsPIC33E" DIR_SEPARATOR_STR \
                                  "inc"
 #define MPLABC30_PIC24H_INC_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "PIC24H" DIR_SEPARATOR_STR \
                                  "inc"
-#define MPLABC30_PIC24E_INC_PATH ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "PIC24E" DIR_SEPARATOR_STR \
-                                 "inc"
 #define MPLABC30_PIC24F_INC_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "PIC24F" DIR_SEPARATOR_STR \
                                  "inc"
 #define MPLABC30_PIC24E_INC_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "PIC24E" DIR_SEPARATOR_STR \
                                  "inc"
@@ -118,58 +122,90 @@
  *   '/' is converted to the correct directory separator for the OS
  */
 #define MPLABC30_COMMON_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "include" PATH_SEPARATOR_STR \
                                      ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "generic" DIR_SEPARATOR_STR \
                                      "h"
 #define MPLABC30_LEGACY_COMMON_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "include" DIR_SEPARATOR_STR "lega-c" \
                                      PATH_SEPARATOR_STR \
                                      ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "generic" DIR_SEPARATOR_STR \
                                      "h"
 #define MPLABC30_PIC24F_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "PIC24F" DIR_SEPARATOR_STR \
                                      "h" PATH_SEPARATOR_STR \
                                      ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "peripheral_24F"
 #define MPLABC30_PIC24H_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "PIC24H" DIR_SEPARATOR_STR \
                                      "h" PATH_SEPARATOR_STR \
                                      ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "peripheral_30F_24H_33F"
 #define MPLABC30_PIC24E_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "PIC24E" DIR_SEPARATOR_STR \
                                      "h" PATH_SEPARATOR_STR \
                                      ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "peripheral_30F_24H_33F"
 #define MPLABC30_PIC30F_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "dsPIC30F" DIR_SEPARATOR_STR \
                                      "h" PATH_SEPARATOR_STR \
                                      ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "peripheral_30F_24H_33F"
 #define MPLABC30_PIC33F_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "dsPIC33F" DIR_SEPARATOR_STR \
                                      "h" PATH_SEPARATOR_STR \
                                      ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "peripheral_30F_24H_33F"
 #define MPLABC30_PIC33E_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "dsPIC33E" DIR_SEPARATOR_STR \
                                      "h" PATH_SEPARATOR_STR \
                                      ".." DIR_SEPARATOR_STR \
+                                     ".." DIR_SEPARATOR_STR \
+                                     "pic30-nonfree" DIR_SEPARATOR_STR \
                                      "support" DIR_SEPARATOR_STR \
                                      "peripheral_30F_24H_33F"
 /*
@@ -178,69 +214,109 @@
  *   '/' is converted to the correct directory separator for the OS
  */
 #define MPLABC30_COMMON_LIB_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "generic" DIR_SEPARATOR_STR \
                                  "gld"
 #define MPLABC30_PIC24F_LIB_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" DIR_SEPARATOR_STR \
                                  "PIC24F" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "PIC24F" DIR_SEPARATOR_STR \
                                  "gld" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" DIR_SEPARATOR_STR \
                                  "peripheral_24F"
 #define MPLABC30_PIC24H_LIB_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" DIR_SEPARATOR_STR \
                                  "PIC24H" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "PIC24H" DIR_SEPARATOR_STR \
                                  "gld" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" DIR_SEPARATOR_STR \
                                  "peripheral_30F_24H_33F"
 #define MPLABC30_PIC24E_LIB_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" DIR_SEPARATOR_STR \
                                  "PIC24E" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "PIC24E" DIR_SEPARATOR_STR \
                                  "gld" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" DIR_SEPARATOR_STR \
                                  "peripheral_30F_24H_33F"
 #define MPLABC30_PIC30F_LIB_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" DIR_SEPARATOR_STR \
                                  "dsPIC30F" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "dsPIC30F" DIR_SEPARATOR_STR \
                                  "gld" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" DIR_SEPARATOR_STR \
                                  "peripheral_30F_24H_33F"
 #define MPLABC30_PIC33F_LIB_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" DIR_SEPARATOR_STR \
                                  "dsPIC33F" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "dsPIC33F" DIR_SEPARATOR_STR \
                                  "gld" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" DIR_SEPARATOR_STR \
                                  "peripheral_30F_24H_33F"
 #define MPLABC30_PIC33E_LIB_PATH ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" DIR_SEPARATOR_STR \
                                  "dsPIC33E" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "support" DIR_SEPARATOR_STR \
                                  "dsPIC33E" DIR_SEPARATOR_STR \
                                  "gld" PATH_SEPARATOR_STR \
                                  ".." DIR_SEPARATOR_STR \
+                                 ".." DIR_SEPARATOR_STR \
+                                 "pic30-nonfree" DIR_SEPARATOR_STR \
                                  "lib" DIR_SEPARATOR_STR \
                                  "peripheral_30F_24H_33F"
 
